Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170. 

In [18]:
import numpy as np
from scipy import stats

#Футболисты
y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182])
# Хоккеисты
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
#Штангисты
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

#Проверка на нормальность (по очереди все)
#stats.shapiro(y1)
#pvalue > alpha -> распределение нормальное

total = np.concatenate((y1, y2, y3))

k = 3
n = y1.size + y2.size + y3.size #28

y1_mean = np.mean(y1)
y2_mean = np.mean(y2)
y3_mean = np.mean(y3)

total_mean = np.mean(total)

S_total = np.sum((total - total_mean)**2)
S_f = np.sum((y1_mean - total_mean)**2) * y1.size + np.sum((y2_mean - total_mean)**2) * y2.size + np.sum((y3_mean - total_mean)**2) * y3.size
S_ost = np.sum((y1 - y1_mean)**2) + np.sum((y2 - y2_mean)**2) + np.sum((y3 - y3_mean)**2)

D_f = S_f / (k - 1)
D_ost = S_ost / (n - k)

F_n = D_f / D_ost
F_n

5.500053450812598

Табличное значение критерия Фишера для k1 = 3 и k2 = 25: 2.99

Соответственно т.к. 5.5 > 2.99 делаем вывод, что различия среднего роста есть
